pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/PythonExtension.groovy

```diff
@@ -129,6 +133,10 @@ class PythonExtension {
         return details
     }
 
+    public void details(Closure cl) {
```

Comment:
I suspect that all we need here is a setter method `setDetails` instead of this public member.
We already have a private class field for details, but it is currently immutable (final) -- hence no setter.
Let's see what others think about this approach.